apps/src/templates/SchoolNotFound.jsx

```diff
@@ -290,9 +272,53 @@ export default class SchoolNotFound extends Component {
     );
   }
 
+}
+
+class GoogleSchoolLocationSearchField extends React.Component {
+  static propTypes = {
+    name: PropTypes.string.isRequired,
+    // If true, this field uses the React controlled input pattern and must
+    // have the value/onChange handlers hooked up to work properly.
+    // If false, is uncontrolled and used only as a submittable <form> element.
+    // @see https://reactpatterns.com/#controlled-input
+    isControlledInput: PropTypes.bool,
+    value: PropTypes.string,
+    onChange: PropTypes.func,
+    style: PropTypes.object,
+  };
+
+  value() {
+    if (this.props.isControlledInput) {
+      return this.props.value;
+    }
+    return this.searchBoxRef.value;
+  }
+
   componentDidMount() {
-    if (this.props.useGoogleLocationSearch) {
-      new google.maps.places.SearchBox(document.getElementById('registration-school-location'));
+    // Docs: https://developers.google.com/maps/documentation/javascript/places-autocomplete#places_searchbox
+    const searchBox = new google.maps.places.SearchBox(this.searchBoxRef);
+    if (this.props.isControlledInput) {
+      searchBox.addListener('places_changed', () => {
+        this.props.onChange({target: this.searchBoxRef});
```

Comment:
If I'm correct, this will be a null ref if the caller forgets to pass the onChange prop - is the presence of that enforceable in some way (conditional on isControlledInput), or is this where we want an error to bubble up from?

Reply:
If we think it will help, I can implement a compound proptypes validation that requires onChange iff isControlledInput is true.